### ISSUE TITLE
With pylint 1.5.0, astroid 1.4.1, pylint-django 0.7, 'get' is not recognised

### ISSUE THREAD

**Issue by jribbens** [landscapeio/pylint-django#53]
The following test script produces an error when linted and it seems to me that it should not:

``` python
"""Test case."""
from django.contrib.auth.models import User
User.objects.get(email="hello", password="there")
```

`E:  3, 0: Instance of 'UserManager' has no 'get' member (no-member)`

(This is with Python 3.4.3, Django 1.8.7.)

**Comment by nedbat** [comment]
I am seeing many more `no-member` violations with pylint==1.5.2, astroid==1.4.2, and pylint-django==0.7.1.

I've written them up as a pylint issue: https://github.com/PyCQA/pylint/issues/754, but perhaps the problem is in the pylint-django plugin?

**Comment by kaleb** [comment]
From the Django [Deserialization of Natural Keys][0] example, the following is also problematic:

    from django.db import models

    class PersonManager(models.Manager):
        # pylint: disable=too-few-public-methods
        # Does pylint not know about subclassing?

        def get_by_natural_key(self, first_name, last_name):
            # pylint: disable=no-member
            # pylint is complaining about no member `get1
            return self.get(first_name=first_name, last_name=last_name)

    class Person(models.Model):
        objects = PersonManager()

        first_name = models.CharField(max_length=100)
        last_name = models.CharField(max_length=100)

        birthdate = models.DateField()

        class Meta:
            unique_together = (('first_name', 'last_name'),)

[0]: https://docs.djangoproject.com/en/1.9/topics/serialization/#deserialization-of-natural-keys

**Comment by mlncn** [comment]
pylint==1.5.0
django==1.7.10
pylint-django==0.7.1
astroid==1.4.4

Just started using pylint-django, updated pylint to be able to do so, it helped with a great many spurious warnings but as noted above, am now getting frequent nomember ones like these:

> Instance of 'Reporter' has no 'save' member (no-member)
> Super of 'ReporterCategory' has no 'save' member (no-member)
> Instance of 'ReporterManager' has no 'get_queryset' member (no-member)
> Instance of 'ReporterManager' has no 'filter' member (no-member)

Which is all to say subscribing, interested in helping.

**Comment by iXce** [comment]
As I faced this problem as well I tried to use different versions of pylint/astroid, and ended up using the git master version of both of them (https://github.com/PyCQA/pylint/commit/9e7bee05d62e682c8199f64c9ae3d787fd5c0072 and https://github.com/PyCQA/astroid/commit/7278409ee6e578c1f40204b062def17d704a08e6).

I then patched pylint-django (https://github.com/iXce/pylint-django):
a) for compatibility with the new api for locals in pylint/astroid: 73326fa
b) considering ForeignObject as an alias of Member (for use with OneToOneFields): f4bcf7dc
c) missing Model/Manager members: 0356359

I'm totally unsure if this was the best way to do it, but this combination of versions works very well for me at the moment =) Hope it can help someone ! And if these approaches are decent, I can do a pullrequest.

**Comment by jribbens** [comment]
@carlio Any chance of getting this issue fixed, given that this whole module is essentially broken currently?

**Comment by nedbat** [comment]
@iXce Please make a pull request.

**Comment by williamboman** [comment]
We had defined

```py
class MyModel(models.Model):
    objects = models.Manager()
```

in a few of our models. Removing these assignments effectively removed pylint errors.
